clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/description/manifest/KubernetesManifest.java

```diff
@@ -241,17 +257,28 @@ public static String getFullResourceName(KubernetesKind kind, String name) {
     return String.join(" ", kind.toString(), name);
   }
 
+  @JsonIgnore
   public boolean isNewerThanObservedGeneration() {
-    try {
-      Long generation = (Long) getMetadata().get("generation");
-      Long observedGeneration = ((Map<String, Long>) getStatus()).get("observedGeneration");
-      if (observedGeneration == null || (generation != null && generation > observedGeneration)) {
-        return false;
-      }
-    } catch (ClassCastException e) {
+    Object generationObj = getMetadata().get("generation");
+    if (generationObj == null) {
+      return false;
+    }
+
+    if (!(generationObj instanceof Integer)) {
+      throw new IllegalStateException("Expected metadata.generation to be an Integer but was actually a " + generationObj.getClass());
+    }
+
+    Integer generation = (Integer) generationObj;
+
+    Integer observedGeneration = getObservedGeneration();
+
+    if (observedGeneration == null) {
+      return false;
     }
-    return true;
+
+    return !(generation > observedGeneration);
```

Comment:
this could be rewritten to `return generation <= observedGeneration`

reading this all again as well, it seems like this method returns true when the observed generation is newer than the manifest generation, even though the name implies that it would return true if the manifest generation is newer than the observed generation?

Reply:
All uses of this method negate the boolean anyway, so it would seem this method was misnamed?